src/main/kotlin/com/terraformation/backend/tracking/model/PlantingZoneModel.kt

```diff
@@ -139,18 +140,44 @@ data class PlantingZoneModel(
     }
   }
 
+  /** Returns true if the zone contains a permanent cluster with the supplied cluster number. */
+  fun permanentClusterExists(clusterNumber: Int): Boolean {
+    return plantingSubzones.any { subzone ->
+      subzone.monitoringPlots.any { it.permanentCluster == clusterNumber }
+    }
+  }
+
+  /**
+   * Returns the highest-valued monitoring plot name from all the plots that have numeric names.
+   * This is used for generating names for new monitoring plots. If there are no monitoring plots,
+   * returns 0.
+   */
+  fun getMaxPlotName(): Int {
+    return plantingSubzones
+        .flatMap { subzone ->
+          subzone.monitoringPlots.mapNotNull { plot -> plot.name.toIntOrNull() }
+        }
+        .maxOrNull()
+        ?: 0
+  }
+
   /**
    * Returns a square Polygon of the requested width/height that is completely contained in the zone
    * and does not intersect with any existing monitoring plots.
    *
    * The square will be positioned a whole multiple of [gridInterval] from the specified origin.
    *
+   * @param excludeAllPermanentPlots If true, exclude the boundaries of all permanent monitoring
+   *   plots. If false, only exclude the boundaries of permanent monitoring plots that will be
+   *   candidates for inclusion in the next observation (that is, ignore permanent plots that were
+   *   used in previous observations but won't be used in the next one).
    * @param exclusion Areas to exclude from the zone, or null if the whole zone is available.
    * @return The unused square, or null if no suitable area could be found.
    */
   fun findUnusedSquare(
       gridOrigin: Point,
-      sizeMeters: Int,
+      sizeMeters: Number,
+      excludeAllPermanentPlots: Boolean = true,
```

Comment:
Just curious, do we have a use case for 'false'?

Reply:
No use for `false` in the context of this PR, but when this gets used to find temporary plots, we will need to be able to pick a plot that's part of a previous permanent cluster that isn't being used in the current observation.